app/controllers/signupAuth.js

```diff
@@ -20,20 +17,24 @@ module.exports.signup = (req, res) => {
     email: req.body.email,
     password: req.body.password
   });
-
-  newUser.save((err, user) => {
+  newUser.save((err, User) => {
     if (err) {
-      res.status(400).json({
+      return res.status(400).json({
         success: false,
         message: 'cannot leave parameter empty'
       });
-    } else {
-      const expires = moment().add(7, 'days').valueOf();
-      const token = jwt.sign({ id: user.id, exp: expires }, 'secret');
-      res.json({
-        success: true,
-        message: 'Successfully created new user.', token, expires
-      });
     }
+    const expires = moment().add(7, 'days').valueOf();
+    const token = jwt.sign({
+      id: User.id,
+      exp: expires
+    }, 'secret');
```

Comment:
I have updated the environment variables with the secret token key so that the build can be possible using travis.